lib/ui.ex

```diff
@@ -1,109 +1,153 @@
 defmodule Ui do
 
   @clear_screen "\e[H\e[2J"
-
-    def ask_for_position(board) do
-      IO.puts "Please choose a position:"
-      get_users_position(board)
-    end
-
-    def get_users_position(board) do
-      input = clean_input(IO.gets(""))
-      case valid_position?(input, board) do
-        {:valid, position} -> position
-        {:taken, position} -> move_was_taken(board, position)
-        :not_a_number -> was_not_a_number(board, input)
-      end
-    end
-
-    defp move_was_taken(board, position) do
-      IO.puts "#{position} is already taken!"
-      ask_for_position(board)
-    end
-
-    defp was_not_a_number(board, input) do
-      IO.puts "#{input} is invalid. We need a number!"
-      ask_for_position(board)
-    end
-
-    def play_again? do
-      IO.puts "Play again? (y/n)"
-      get_users_replay_choice
-    end
-
-    def print_board(board) do
-      clear_screen
+  @game_mode [
+    {1, "Human vs Human", :human_vs_human},
+    {2, "Human vs Random", :human_vs_random},
+    {3, "Random vs Human", :random_vs_human},
+  ]
+
+  def ask_for_game_mode do
+    clear_screen
+    IO.puts "Please choose a game mode:\n"
+    Enum.map(@game_mode, fn({number, option, _}) ->
+      IO.puts "#{number} - #{option}"
```

Comment:
Well, I'm using it as a way to iterate over the list and print out each line... But I've changed it now to do it recursively instead.